cogs/nerd_ecke.py

```diff
@@ -0,0 +1,106 @@
+from typing import Union
+
+from discord import Member, User, VoiceState, TextChannel, Embed, Role, VoiceChannel, Guild
+from discord.ext.commands import Cog, Bot, group, has_guild_permissions, Context, BadArgument
+from discord.ext.tasks import loop
+
+from cogs.bot_status import listener
+from cogs.util.ainit_ctx_mgr import AinitManager
+from cogs.util.assign_variables import assign_chat
+from cogs.util.placeholder import Placeholder
+from core.global_enum import CollectionEnum, ConfigurationNameEnum
+from core.logger import get_discord_child_logger
+from core.predicates import bot_chat
+from mongo.primitive_mongo_data import PrimitiveMongoData
+
+bot_channels: set[TextChannel] = set()
+nerd_channel = Placeholder()
+first_init = True
+
+logger = get_discord_child_logger("Nerd-Ecke")
+
+
+class NerdEcke(Cog):
+    """
+    Small collection of commands for various purposes.
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot: Bot = bot
+        self.need_init = True
+        if not first_init:
+            self.ainit.start()
+
+    @loop()
+    async def ainit(self):
+        """
+        Loads the configuration for the module.
+        """
+        global bot_channels, nerd_channel
+        # noinspection PyTypeChecker
+        async with AinitManager(bot=self.bot,
+                                loop=self.ainit,
+                                need_init=self.need_init,
+                                bot_channels=bot_channels) as need_init:
+            if need_init:
+                nerd_channel.item = await assign_chat(self.bot, ConfigurationNameEnum.NERD_VOICE_CHANNEL)
+
+    @listener()
+    async def on_ready(self):
+        global first_init
+        if first_init:
+            first_init = False
+            self.ainit.start()
+
+    @listener()
+    async def on_voice_state_update(self, member: Union[Member, User], before: VoiceState, after: VoiceState):
+        if before.channel == nerd_channel.item or after.channel == nerd_channel.item:
+            role: Role = member.guild.default_role
+
+            members: set[Union[Member, User]] = {member for member in nerd_channel.item.members if not member.bot}
+
+            if members:
+                await nerd_channel.item.set_permissions(role, connect=True, reason="Nerd is here.")
+            else:
+                await nerd_channel.item.set_permissions(role, connect=False, reason="No nerds are here.")
+
+    @group(pass_context=True,
+           name="nerd")
+    @bot_chat(bot_channels)
+    @has_guild_permissions(administrator=True)
+    async def nerd(self, ctx: Context):
+        if not ctx.invoked_subcommand:
+            raise BadArgument
+        member: Union[Member, User] = ctx.author
+        logger.info(f'User="{member.name}#{member.discriminator}({member.id})", Command="{ctx.message.content}"')
+
+    @nerd.command(pass_context=True,
+                  name="voice")
+    async def study_channel_voice(self, ctx: Context, channel: int):
```

Comment:
Maybe a better method name?
nerd_voice or something?

But it doesn't have to be.

Reply:
I mean it is a admin only command that is used once at the beginning should be fine